### PR TITLE
[Issue 9804] Allow to enable or disable the cursor metrics

### LINKED ISSUE #9804: Allow to enable or disable the cursor metrics in the broker.conf
**Is your feature request related to a problem? Please describe.**

https://github.com/apache/pulsar/pull/9618 introduced the managed cursor metrics, it's better to add a flag in the broker.conf to enable or disable the cursor level metrics.

`exposeManagedCursorMetricsInPrometheus=false`, keep the default value as false.

### PR DESCRIPTION
**Pull request by linlinnn**
Fixes #9804 

### Motivation
Allow to enable or disable the cursor metrics

### Modifications
add config item `exposeManagedCursorMetricsInPrometheus`  to allow user to enable or disable the cursor metrics

### Verifying this change
- [ ] Make sure that the change passes the CI checks.

### CODE REVIEW

**Review by codelipenghui — COMMENTED** [review]
Could you please add some tests?

**Comment by codelipenghui** [comment]
@limingnihao Please help review this PR, thanks.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java [hunk 1]
```diff
@@ -130,9 +130,11 @@ private static void generateBrokerBasicMetrics(PulsarService pulsar, SimpleTextO
         parseMetricsToPrometheusMetrics(new ManagedLedgerMetrics(pulsar).generate(),
                 clusterName, Collector.Type.GAUGE, stream);
 
-        // generate managedCursor metrics
-        parseMetricsToPrometheusMetrics(new ManagedCursorMetrics(pulsar).generate(),
-                clusterName, Collector.Type.GAUGE, stream);
+        if(includeManagedCursorMetrics){
```
Comment (murong00) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java:133]: Why not use `pulsar.getConfiguration().isExposeManagedCursorMetricsInPrometheus()
` here? please note the code style `if (...) {}`.

**Review by murong00 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java (hunk 1 above)
Comment (linlinnn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java:133]: Just for being consistent with `includeConsumerMetrics`, `includeProducerMetrics` etc.

**Review by limingnihao — APPROVED** [review]

**Review by MarvinCai — APPROVED** [review]
LGTM, please add a test case or update existing test case in PrometheusMetricsTest to verify the behavior.

**Review by eolivelli — CHANGES_REQUESTED** [review]
LGTM

Please add a testcase

**Review by eolivelli — APPROVED** [review]
Looks great

Thanks

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@murong00 Could you please review this PR again?

**Review by murong00 — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**